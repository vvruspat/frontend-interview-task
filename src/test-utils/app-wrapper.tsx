import React, { FC } from "react";
import { Provider } from "react-redux";
import { GlobalStyles } from "../components/global-styles";
import { AuthProvider } from "../auth";
import { store } from "../store/store";

export const AppWrapper: FC = ({ children }) => {
  return (
    <Provider store={store}>
      <GlobalStyles />
      <AuthProvider>{children}</AuthProvider>
    </Provider>
  );
};
