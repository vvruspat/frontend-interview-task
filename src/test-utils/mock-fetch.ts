import { error401Data } from "./test-data/error";
import { feedbacksData } from "./test-data/feedbacks";
import { jwtData } from "./test-data/jwt";
import { themesData } from "./test-data/themes";

export function mockFetch() {
  let status: number = 200;

  return {
    setStatus: (newStatus: number) => {
      status = newStatus;
    },

    fetch: jest.fn().mockImplementation((url: string) => {
      if (status !== 200) {
        return Promise.reject(error401Data);
      }

      if (url.indexOf("/api/reviews") !== -1) {
        return Promise.resolve({
          ok: true,
          status: status,
          json: () => Promise.resolve({ data: feedbacksData }),
        });
      }

      if (url.indexOf("/api/themes") !== -1) {
        return Promise.resolve({
          ok: true,
          status: status,
          json: () => Promise.resolve({ data: themesData }),
        });
      }

      if (url.indexOf("/login") !== -1) {
        return Promise.resolve({
          ok: true,
          status: status,
          json: () => Promise.resolve(jwtData),
        });
      }

      return Promise.resolve({
        ok: true,
        status: status,
        json: () => {},
      });
    }),
  };
}
