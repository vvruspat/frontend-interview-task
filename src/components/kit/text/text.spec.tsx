import React from "react";
import { render } from "@testing-library/react";
import { AppWrapper } from "../../../test-utils/app-wrapper";
import { Text } from ".";

describe("text", () => {
  test("renders text without props", async () => {
    const component = render(<Text>Test</Text>, { wrapper: AppWrapper });
    const text = component.getByText("Test");

    expect(text).toMatchSnapshot();
  });

  test("renders text centered", async () => {
    const component = render(<Text center={true}>Test</Text>, {
      wrapper: AppWrapper,
    });
    const text = component.getByText("Test");

    expect(text).toMatchSnapshot();
  });

  test("renders text big", async () => {
    const component = render(<Text size={"large"}>Test</Text>, {
      wrapper: AppWrapper,
    });
    const text = component.getByText("Test");

    expect(text).toMatchSnapshot();
  });

  test("renders text small", async () => {
    const component = render(<Text size={"small"}>Test</Text>, {
      wrapper: AppWrapper,
    });
    const text = component.getByText("Test");

    expect(text).toMatchSnapshot();
  });
});
