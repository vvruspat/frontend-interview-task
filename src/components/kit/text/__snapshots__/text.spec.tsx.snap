// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text renders text big 1`] = `
<div
  class="sc-bdfBwQ gfzQwd"
>
  Test
</div>
`;

exports[`text renders text centered 1`] = `
<div
  class="sc-bdfBwQ ZOXct"
>
  Test
</div>
`;

exports[`text renders text small 1`] = `
<div
  class="sc-bdfBwQ ksNmpp"
>
  Test
</div>
`;

exports[`text renders text without props 1`] = `
<div
  class="sc-bdfBwQ lnvQCl"
>
  Test
</div>
`;
